module3/lessons/intro-to-cypress-prework.md

```diff
@@ -0,0 +1,437 @@
+
+* First use [the Cypress docs](https://docs.cypress.io/guides/getting-started/installing-cypress.html#Installing){:target='blank'} to figure out how to use npm to install Cypress within the Feedback Loop UI repo.
+* There are multiple ways of opening up Cypress.  Use the docs to figure out how to setup a `script` in the `package.json` that you'll be able to use to open up Cypress.  
+* Open Cypress with the script you added. A new window will appear with two testing options.  
+  * Select **E2E Testing**.  
+  * There are some config options on the next window...but for now just select **Continue** at the bottom.  
+    * *Hint: If you see an error that references webpack, you likely chose Component Testing by mistake in the previous step.*
+  * Then select your browser (*Chrome*) and the **Start E2E Testing in Chrome** button. *Pause here for now and keep working thru this prework lesson.*  
+
+Having completed these steps, you should notice some new directories and files added to your application.
+
+
+<section class="dropdown">
+### The Answer
+
+If you get stuck!  Look at the Cypress docs or our IdeaBox example.
+</section>
+
+
+### Writing your first test!
+You'll notice in your code that there are a few directories including `downloads`, `fixtures`, and `support` inside of a `cypress` directory.  Make note of these directories and go back to the main window on Cypress, then follow the [instructions here](https://docs.cypress.io/guides/end-to-end-testing/writing-your-first-end-to-end-test#What-you-ll-learn){:target="_blank"} for adding your first test file.
+
+### Creating some test files
+As we consider what we will be testing, let's consider a few ways to set up our files.
+
+We could make one giant file and test absolutely everything there: `feedback_loop_spec.cy.js`.  But it's probably more maintainable to group up our related user flows.
+
+**Create a few files in the `e2e` directory (located inside the `cypress` directory) using the Cypress UI** (do not add them manually within your repo):
+- `cypress/e2e/login_spec.cy.js`
+- `cypress/e2e/dashboard_spec.cy.js`
+- `cypress/e2e/form_spec.cy.js`
+
+<section class="note">
+
+
+Notice that each of these describes actions tied to our data/server/network requests. When viewing feedback from coworkers, there are several different user flows. But they all involve GETTING feedback data from the back end.
+
+Figuring out how to group user flows/stories can be tricky, and ultimately there are no hard-and-fast rules about how to do so. Over time, you'll develop a sense of what to put together, just like how you are learning what to break out into a React component and what to leave as is. And, of course, these conventions change from team to team.
+</section>
+
```

Comment:
The deliverable states that they need to submit a screenshot of their cypress test runner after making a test to test that the form is there on page load.   But the instructions don't ask them to do that.  Since the actual lesson's first coding exercise has them test for the form display on page load, maybe we dont have that be the deliverable.

As long as they have cypress installed and the three files made in the e2e directory, I think thats ok.   Maybe we just have them submit a screenshot of their e2e directory with the files?